doc/rst/source/devdocs/contributing.rst

```diff
@@ -239,16 +239,23 @@ You may need to know some basic reST syntax before making changes. Please refer
 Building the documentation
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Usually you don't need to build the documentation locally for small changes. To build the GMT documentation you need
-to install the `Sphinx <http://www.sphinx-doc.org/>`_ documentation builder. After configuring and
-`building GMT <https://github.com/GenericMappingTools/gmt/tree/master/BUILDING.md>`_, you can build GMT documentation using
-the following commands within the build directory::
+Usually you don't need to build the documentation locally for small changes. To build the GMT documentation you 
+need to `build GMT from source <https://github.com/GenericMappingTools/gmt/tree/master/BUILDING.md>`_. Be sure 
+to also satisfy the 
+`developement dependencies <https://github.com/GenericMappingTools/gmt/tree/master/BUILDING.md#development-dependencies>`_ 
+before proceeding. Have a look at the options in ``cmake/ConfigUserAdvanced.cmake`` if you want to change the 
+target directory for the documenation you are about to build.
+
+After `configuring and building GMT from source <https://github.com/GenericMappingTools/gmt/tree/master/BUILDING.md>`_, 
+you can then build the GMT documentation using the following commands within the ``build`` directory::
 
   dvc pull
   cmake --build . --target docs_depends     # Generate images included in the documentation
   cmake --build . --target optimize_images  # Optimize PNG images for documentation [optional]
-  cmake --build . --target docs_man         # UNIX manual pages
-  cmake --build . --target docs_html        # HTML manual, tutorial, cookbook, and API reference
+  cmake --build . --target docs_man         # Build UNIX manual pages
+  cmake --build . --target docs_html        # Build HTML manual, tutorial, cookbook, and API reference
+
+  cmake --build . --target install          # install the documentation into the specified location
```

Comment:
```suggestion
To install the UNIX manpages and html documentation into the specified location (along with the gmt executable, library, development headers and built-in data), use::

  cmake --build . --target install
```
I think it's helpful to specify that this command does more than just installing the documentation

Reply:
That's a good suggestion @maxrjones. Before I commit it maybe you have some deeper knowledge on the doc build & install process:

* All plots are unnecessarily generated from scratch every time (time consuming). 
Is there a way around that? I don't really need to create the plots from scratch every time when I just changed some links, layout related things or fixed a typo. Sure, `--parallel` does help but still wasted time.
* No apparent way of installing just the documentation.
It is not as time consuming as building but still not really necessary to install everything every time. 

I think that would make working on the documentation a bit more convenient.

Reply:
These are good points @KristofKoch. unfortunately I haven't had time to look into options. Perhaps you could raise this in an issue so someone (maybe me) can look into the CMake setup later on?

Reply:
> * All plots are unnecessarily generated from scratch every time (time consuming).
>   Is there a way around that? I don't really need to create the plots from scratch every time when I just changed some links, layout related things or fixed a typo. Sure, `--parallel` does help but still wasted time.

I think you don't generate the plots from scratch. The plots were saved as PS files in directories like `doc/scripts/images`. Now, these PS files are stored on the dvc remote server and can be downloaded to your computer using `dvc pull`. When building the documentation, it still needs to convert PS images to PNG images.

BTW, if you don't remove your `build` directory, running commands like `cmake --build . --target docs_html` will only rebuild the files you have changed, so it's much faster.

> * No apparent way of installing just the documentation.
>   It is not as time consuming as building but still not really necessary to install everything every time.

Could you please try if the following command works as you expect?
```
cmake --build . -DCOMPONENT=Documentation --target install
```

Reply:
Hi @seisman I just created a new feature request (#7225) for that as recommended by @maxrjones … just missed your comment by some minutes. Would you mind shifting the discussion over there?

Your command didn't work for me:

```
% cmake --build . --target docs_html --parallel
[…]
% cmake --build . -DCOMPONENT=Documentation --target install
Unknown argument -DCOMPONENT=Documentation
```